src/app/cartography/tools/text-editing-tool.ts

```diff
@@ -34,45 +41,44 @@ export class TextEditingTool {
 
             this.editingDrawingId = textElements[index].parentElement.parentElement.getAttribute("drawing_id");
 
-            select(textElements[index].parentElement.parentElement.parentElement)
-                .append("foreignObject")
-                .attr("width", '1000px')
-                .attr("min-width", 'fit-content')
-                .attr("height", '100px')
-                .attr("id", "temporaryText")
-                .attr("transform", textElements[index].parentElement.getAttribute("transform"))
-                .append("xhtml:span")
-                .attr("width", "fit-content")
-                .attr("height", "fit-content")
-                .attr("class", "temporaryTextInside")
-                .attr('style', () => {
-                    const styles: string[] = [];
-                    styles.push(`white-space: pre-line`)
-                    styles.push(`outline: 0px solid transparent`)
-                    styles.push(`font-family: ${elem.font_family}`)
-                    styles.push(`font-size: ${elem.font_size}pt!important`);
-                    styles.push(`font-weight: ${elem.font_weight}`)
-                    styles.push(`color: ${elem.fill}`);
-                    return styles.join("; ");
-                })
-                .attr('text-decoration', elem.text_decoration)
-                .attr('contenteditable', 'true')
-                .text(elem.text)
-                .on("focusout", () => {
-                    let temporaryText = document.getElementsByClassName("temporaryTextInside")[0] as HTMLElement;
-                    let savedText = temporaryText.innerText;
-                    this.editingFinished.emit(new TextEditedDataEvent(this.editingDrawingId, savedText, this.editedElement));
+            var transformData = textElements[index].parentElement.getAttribute("transform").split(/\(|\)/);
+            var x = Number(transformData[1].split(/,/)[0]) + this.context.getZeroZeroTransformationPoint().x;
+            var y = Number(transformData[1].split(/,/)[1]) + this.context.getZeroZeroTransformationPoint().y;
 
-                    var temporaryElement = document.getElementById("temporaryText") as HTMLElement;
-                    temporaryElement.remove();
+            this.temporaryElement = document.createElement('div');
```

Comment:
I think we need to decide if we use D3.js or Angular's approach. Definitely we should try to not access DOM directly.